File: python-single/src/klv/common/enums/meterSource.py
from enum import IntEnum


class MeterSource(IntEnum):
  MAIN = 0
  GROUP = 1
  AUX = 2
  TRACK = 3
  EXTERNAL_INPUT = 4
  MIX_MINUS = 5
  AFL_STEREO = 6
  AFL_SURROUND = 7
  PFL_STEREO = 8
  PFL_SURROUND = 9
  APFL_STEREO = 10
  APFL_SURROUND = 11
  CONSOLE_LS = 12
  OAC = 13
  USER_METER_1 = 14
  USER_METER_2 = 15
  FADER = 16
  VIDEO = 17

  def getFullTitle(self):
    return "TODO"

  def getLabelSuffix(self):
    return "TODO"
  
  def isMixMinusSource(self):
    return self is MeterSource.MIX_MINUS

  def getMixMinusMeterSourceByIndex(self, _faderIndexNumber: int):
    if self.isMixMinusSource():
      # TODO
      # this.fullTitle = "Mix Minus" + " " + index;
      # this.index = index;
      pass
    return self

File: rust-single/src/klv/common/enums/meterSource.rs
#![allow(non_snake_case, non_camel_case_types)]

#[derive(Debug, Copy, Clone)]
pub enum MeterSource {
  MAIN,
  GROUP,
  AUX,
  TRACK,
  EXTERNAL_INPUT,
  MIX_MINUS,
  AFL_STEREO,
  AFL_SURROUND,
  PFL_STEREO,
  PFL_SURROUND,
  APFL_STEREO,
  APFL_SURROUND,
  CONSOLE_LS,
  OAC,
  USER_METER_1,
  USER_METER_2,
  FADER,
  VIDEO,
}

impl MeterSource {
  pub fn getFullTitle(self) -> String {
    "TODO".into()
  }

  pub fn getLabelSuffix(self) -> String {
    "TODO".into()
  }
  
  pub fn isMixMinusSource(self) -> bool {
    matches!(self, MeterSource::MIX_MINUS)
  }

  pub fn getMixMinusMeterSourceByIndex(self, _faderIndexNumber: i32) -> MeterSource {
    if self.isMixMinusSource() {
      // TODO
      // this.fullTitle = "Mix Minus" + " " + index;
      // this.index = index;
    }
    return self.clone();
  }
}